AssertionError in validated_integ if rhs has a sine term

```julia
using Test
using TaylorModels
using TaylorModels: validated_integ

function pendulum!(dx, x, p, t)
    aux = 2 * sin(x[1])
    dx[1] = x[2]
    dx[2] = aux + 8*x[3]
    dx[3] = zero(x[1])
    return nothing
end

# Initial conditions
tini, tend = 0.0, 0.03
q0 = [1.2, 0.2, -0.78]
δq0 = IntervalBox(0 .. 0, 3)
X0 = IntervalBox(q0 .+ δq0)

# Parameters
abstol = 1e-20
orderQ = 3
orderT = 10
ξ = set_variables("ξ", order=2*orderQ, numvars=length(q0))

normalized_box = IntervalBox(-1 .. 1, Val(orderQ))

tTM, qv, qTM = validated_integ(pendulum!, X0, tini, tend, orderQ, orderT, abstol)

AssertionError: get_numvars() == length(vals)

Stacktrace:
  [1] _evaluate(a::TaylorN{Interval{Float64}}, vals::Tuple{Interval{Float64}})
    @ TaylorSeries ~/.julia/packages/TaylorSeries/tveWm/src/evaluate.jl:229
  [2] _evaluate
    @ ~/.julia/packages/TaylorSeries/tveWm/src/evaluate.jl:240 [inlined]
  [3] evaluate(a::TaylorN{Interval{Float64}}, vals::Interval{Float64}; sorting::Bool)
    @ TaylorSeries ~/.julia/packages/TaylorSeries/tveWm/src/evaluate.jl:223
  [4] evaluate
    @ ~/.julia/packages/TaylorSeries/tveWm/src/evaluate.jl:223 [inlined]
  [5] _broadcast_getindex_evalf
    @ ./broadcast.jl:648 [inlined]
  [6] _broadcast_getindex
    @ ./broadcast.jl:621 [inlined]
  [7] getindex
    @ ./broadcast.jl:575 [inlined]
  [8] macro expansion
    @ ./broadcast.jl:984 [inlined]
  [9] macro expansion
    @ ./simdloop.jl:77 [inlined]
 [10] copyto!
    @ ./broadcast.jl:983 [inlined]
 [11] copyto!
    @ ./broadcast.jl:936 [inlined]
 [12] copy
    @ ./broadcast.jl:908 [inlined]
 [13] materialize
    @ ./broadcast.jl:883 [inlined]
 [14] broadcasted
    @ ~/.julia/packages/IntervalArithmetic/dhnJB/src/multidim/arithmetic.jl:29 [inlined]
 [15] picard_remainder!(f!::typeof(pendulum!), t::Taylor1{Float64}, x::Vector{Taylor1{TaylorN{Float64}}}, dx::Vector{Taylor1{TaylorN{Float64}}}, xxI::Vector{Taylor1{TaylorN{Interval{Float64}}}}, dxxI::Vector{Taylor1{TaylorN{Interval{Float64}}}}, δI::IntervalBox{3, Float64}, δt::Interval{Float64}, Δx::IntervalBox{3, Float64}, Δ0::IntervalBox{3, Float64}, params::Nothing)
    @ TaylorModels ~/.julia/dev/TaylorModels/src/validatedODEs.jl:108
 [16] remainder_taylorstep!(f!::Function, t::Taylor1{Float64}, x::Vector{Taylor1{TaylorN{Float64}}}, dx::Vector{Taylor1{TaylorN{Float64}}}, xI::Vector{Taylor1{Interval{Float64}}}, dxI::Vector{Taylor1{Interval{Float64}}}, δI::IntervalBox{3, Float64}, δt::Interval{Float64}, params::Nothing)
    @ TaylorModels ~/.julia/dev/TaylorModels/src/validatedODEs.jl:32
 [17] validated_step!(f!::typeof(pendulum!), t::Taylor1{Float64}, x::Vector{Taylor1{TaylorN{Float64}}}, dx::Vector{Taylor1{TaylorN{Float64}}}, xaux::Vector{Taylor1{TaylorN{Float64}}}, tI::Taylor1{Float64}, xI::Vector{Taylor1{Interval{Float64}}}, dxI::Vector{Taylor1{Interval{Float64}}}, xauxI::Vector{Taylor1{Interval{Float64}}}, t0::Float64, tmax::Float64, sign_tstep::Int64, xTMN::Vector{TaylorModelN{3, Float64, Float64}}, xv::Vector{IntervalBox{3, Float64}}, rem::Vector{Interval{Float64}}, zbox::IntervalBox{3, Float64}, symIbox::IntervalBox{3, Float64}, nsteps::Int64, orderT::Int64, abstol::Float64, params::Nothing, parse_eqs::Bool, check_property::TaylorModels.var"#36#38")
    @ TaylorModels ~/.julia/dev/TaylorModels/src/validatedODEs.jl:361
 [18] validated_integ(f!::Function, X0::IntervalBox{3, Float64}, t0::Float64, tmax::Float64, orderQ::Int64, orderT::Int64, abstol::Float64, params::Nothing; maxsteps::Int64, parse_eqs::Bool, check_property::Function)
    @ TaylorModels ~/.julia/dev/TaylorModels/src/validatedODEs.jl:533
 [19] validated_integ (repeats 2 times)
    @ ~/.julia/dev/TaylorModels/src/validatedODEs.jl:481 [inlined]
 [20] top-level scope
    @ In[15]:27
 [21] eval
    @ ./boot.jl:360 [inlined]
 [22] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
    @ Base ./loading.jl:1094
```

-----

Let me remark that for `aux = 2` there is no error.

--- a/Project.toml
+++ b/Project.toml
@@ -1,7 +1,7 @@
 name = "TaylorModels"
 uuid = "314ce334-5f6e-57ae-acf6-00b6e903104a"
 repo = "https://github.com/JuliaIntervals/TaylorModels.jl.git"
-version = "0.3.9"
+version = "0.3.10"
 
 [deps]
 IntervalArithmetic = "d1acc4aa-44c8-5952-acd4-ba5d80a2a253"

--- a/src/validatedODEs.jl
+++ b/src/validatedODEs.jl
@@ -105,7 +105,7 @@ function picard_remainder!(f!::Function, t::Taylor1{T},
     f!(dxxI, xxI, params, t)
 
     # Picard iteration, considering only the bound of `f` and the last coeff of f
-    Δdx = IntervalBox( evaluate.( (dxxI - dx)(δt), δI ) )
+    Δdx = IntervalBox( evaluate.( (dxxI - dx)(δt), (δI,) ) )
     Δ = Δ0 + Δdx * δt
     return Δ
 end

--- a/src/validatedODEs.jl
+++ b/src/validatedODEs.jl
@@ -540,14 +540,7 @@ function validated_integ(f!, X0, t0::T, tmax::T, orderQ::Int, orderT::Int, absto
     @inbounds xv[1] = evaluate(xTMN, S)
 
     # Determine if specialized jetcoeffs! method exists (built by @taylorize)
-    parse_eqs = parse_eqs && (length(methods(TaylorIntegration.jetcoeffs!)) > 2)
-    if parse_eqs
-        try
-            TaylorIntegration.jetcoeffs!(Val(f!), t, x, dx, params)
-        catch
-            parse_eqs = false
-        end
-    end
+    parse_eqs = TaylorIntegration._determine_parsing!(parse_eqs, f!, t, x, dx, params)
 
     # Integration
     nsteps = 1
@@ -752,10 +745,9 @@ function validated_integ2(f!, X0, t0::T, tf::T, orderQ::Int, orderT::Int,
         δt = sign_tstep * δt
 
         @inbounds for i in eachindex(x)
-            dom = sign_tstep > 0 ? Interval(0, δt) : Interval(δt, 0)
-            x0 = sign_tstep > 0 ? dom.lo : dom.hi
-            Δ = zero(Interval{Float64})
-            xTM1[i] = TaylorModel1(deepcopy(x[i]), Δ, x0, dom)
+            dom = sign_tstep * Interval{T}(0, sign_tstep*δt)
+            Δ = zero(dom)
+            xTM1[i] = TaylorModel1(deepcopy(x[i]), Δ, zI, dom)
         end
         
         # to reuse the previous TaylorModel and save some allocations

--- a/test/validated_integ.jl
+++ b/test/validated_integ.jl
@@ -235,4 +235,52 @@ interval_rand(X::IntervalBox) = interval_rand.(X)
             end
         end
     end
+
+    @testset "Pendulum with constant torque" begin
+        @taylorize function pendulum!(dx, x, p, t)
+            si = sin(x[1])
+            aux = 2 *  si
+            dx[1] = x[2]
+            dx[2] = aux + 8*x[3]
+            dx[3] = zero(x[1])
+            nothing
+        end
+        # Conseerved quantity
+        ene_pendulum(x) = x[2]^2/2 + 2 * cos(x[1]) - 8 * x[3]
+
+        # Initial conditions
+        tini, tend = 0.0, 12.0
+        q0 = [1.1, 0.1, 0.0]
+        δq0 = IntervalBox(-0.1 .. 0.1, -0.1 .. 0.1, 0..0)
+        X0 = IntervalBox(q0 .+ δq0)
+        ene0 = ene_pendulum(X0)
+
+        # Parameters
+        abstol = 1e-20
+        orderQ = 3
+        orderT = 10
+        ξ = set_variables("ξ", order=2*orderQ, numvars=length(q0))
+
+        tTM, qv, qTM = validated_integ(pendulum!, X0, tini, tend, orderQ, orderT, abstol,
+            maxsteps=1800);
+        @test all(ene0 .⊆ ene_pendulum.(qv))
+
+        # tTM, qv, qTM = validated_integ2(pendulum!, X0, tini, tend, orderQ, orderT, abstol,
+        #     maxsteps=1800);
+        # @test all(ene0 .⊆ ene_pendulum.(qv))
+
+        # Initial conditions 2
+        q0 = [1.1, 0.1, 0.0]
+        δq0 = IntervalBox(-0.1 .. 0.1, -0.1 .. 0.1, -0.01 .. 0.01)
+        X0 = IntervalBox(q0 .+ δq0)
+        ene0 = ene_pendulum(X0)
+
+        tTM, qv, qTM = validated_integ(pendulum!, X0, tini, tend, orderQ, orderT, abstol,
+            maxsteps=1800);
+        @test all(ene0 .⊆ ene_pendulum.(qv))
+
+        # tTM, qv, qTM = validated_integ2(pendulum!, X0, tini, tend, orderQ, orderT, abstol,
+        #     maxsteps=1800);
+        # @test all(ene0 .⊆ ene_pendulum.(qv))
+    end
 end

